sirepo/package_data/static/js/genesis.js

```diff
@@ -56,6 +59,32 @@ SIREPO.app.directive('appFooter', function() {
     };
 });
 
+SIREPO.app.directive('maginFilePlot', function(appState) {
+    return {
+        restrict: 'A',
+        scope: {
+            modelName: '@',
+        },
+        template: `
+            <div data-ng-if="hasMaginfile">
+              <div data-show-loading-and-error="" data-model-key="{{ modelName }}">
+                <div data-parameter-plot="parameter" data-model-name="{{ modelName }}"></div>
+              </div>
+            </div>
+        `,
+        controller: function($scope) {
+            const checkForMaginfile = () => {
+                return appState.models.io.maginfile || false;
+            };
+
+            $scope.hasMaginfile = checkForMaginfile();
+            $scope.$on('io.changed', ()=> {
+                $scope.hasMaginfile = checkForMaginfile();
+            });
+        }
```

Comment:
You could replace all the controller code by adding: 
```javascript 
    $scope.hasMaginfile = () => appState.applicationState().io.maginfile;
```
And changing the line to `data-ng-if="hasMaginfile()"`

Also, add a `data-report-id="reportId"` to the parameter-plot, and `$scope.reportId = utilities.reportId();` so you get the popup menu when you click on the plot.

Reply:
<img width="240" alt="Screenshot 2024-02-09 at 4 00 48 PM" src="https://github.com/radiasoft/sirepo/assets/49498231/63016d5e-a676-48ec-b887-e1ff45807f17">
This popup?